scripts/pre-install.js

```diff
@@ -169,30 +183,45 @@ if (platform === 'win32' && os.arch() !== 'ia32') {
     process.exit(1);
 }
 
-// Check if nrfjprog libraries are working or not
+let isInstallationRequired = false;
 getLibraryVersion()
     .then(version => {
-        console.log('Found nrfjprog libraries at version', version);
+        if (version.major < REQUIRED_VERSION.major ||
+            version.minor < REQUIRED_VERSION.minor ||
+            version.revision < REQUIRED_VERSION.revision) {
+            console.log(`Found nrfjprog version ${JSON.stringify(version)}, ` +
+                `but ${JSON.stringify(REQUIRED_VERSION)} is required`);
+            isInstallationRequired = true;
+        } else {
+            console.log('Found nrfjprog libraries at required version', version);
+        }
     })
-    .catch(() => {
-        // If we have nrfjprog header files on win32, then assume nrfjprog is installed.
+    .catch(error => {
+        // If we have nrfjprog header files on win32, then assume nrfjprog is installed
         if (platform === 'win32' && isHeaderFileInstalledWin32()) {
-            return Promise.resolve();
+            isInstallationRequired = false;
+        } else {
+            console.log(`Validation of nrfjprog libraries failed: ${error.message}`);
+            isInstallationRequired = true;
         }
-
-        console.log('Could not find nrfjprog libraries. Trying to install.');
-
-        let exitCode = 0;
-        return downloadFile(platformConfig.url, platformConfig.destinationFile)
-            .then(() => installNrfjprog(platformConfig.destinationFile))
-            .catch(error => {
-                exitCode = 1;
-                console.error(`Error when installing nrfjprog libraries: ${error.message}`);
-            })
-            .then(() => removeFileIfExists(platformConfig.destinationFile))
-            .catch(error => {
-                exitCode = 1;
-                console.error(`Unable to remove downloaded nrfjprog artifact: ${error.message}`);
-            })
-            .then(() => process.exit(exitCode));
+    })
+    .then(() => {
+        if (isInstallationRequired) {
+            console.log('Trying to install nrfjprog');
+
+            let exitCode = 0;
+            return downloadFile(platformConfig.url, platformConfig.destinationFile)
+                .then(() => installNrfjprog(platformConfig.destinationFile))
+                .catch(error => {
+                    exitCode = 1;
+                    console.error(`Error when installing nrfjprog libraries: ${error.message}`);
+                })
+                .then(() => removeFileIfExists(platformConfig.destinationFile))
+                .catch(error => {
+                    exitCode = 1;
+                    console.error(`Unable to remove downloaded nrfjprog artifact: ${error.message}`);
+                })
+                .then(() => process.exit(exitCode));
+        }
+        return Promise.resolve();
```

Comment:
I believe this line (return Promise) could be removed and the effect would be the same. The *then* would resolve in the same way, right?